src/wasm/wasm-stack.cpp

```diff
@@ -1459,6 +1459,10 @@ void BinaryInstWriter::visitBinary(Binary* curr) {
     case MaxUVecI32x4:
       o << int8_t(BinaryConsts::SIMDPrefix) << U32LEB(BinaryConsts::I32x4MaxU);
       break;
+    case DotSVecI16x8ToVecI32x4:
+      o << int8_t(BinaryConsts::SIMDPrefix)
+        << U32LEB(BinaryConsts::I32x4DotSVecI16x8);
+      break;
```

Comment:
It's not particularly about thie PR but Just curious, is there a reason we have different names for Op and BinaryConsts, such as `ExtractLaneSVecI18x16` vs. `I8x16ExtractLaneS`?

Reply:
The BinaryConsts exactly mirror the underlying instruction naming with `.` and `_` removed and the name CamelCased, but the Binaryen Op names are shuffled around to put the operation first and the relevant types later. This name change is meant to increase clarity by emphasizing the operation over the types, but it can also make the types clearer in conversion ops by inserting `To` into the op name, as in `DotSVecI16x8ToVecI32x4`.